src/encoded/tests/test_download_cli.py

```diff
@@ -0,0 +1,97 @@
+import pytest
+from ..download_cli import extract_bucket_and_key
+
+
+@pytest.mark.parametrize('url, expected', [
+    ('https://test-bucket.s3.com/file.bam',
+     ('test-bucket', 'file.bam')),
+    ('http://test-bucket.s3.amazonaws.com/file.bam',
+     ('test-bucket', 'file.bam')),
+    ('https://test-bucket.s3.amazonaws.com/file.bam',
+     ('test-bucket', 'file.bam')),
+    ('https://smaht-unit-testing-wfout.s3.amazonaws.com/cca15caa-bc11-4a6a-8998-ea0c69df8b9d/TSTFI0211073.bam',
+     ('smaht-unit-testing-wfout', 'cca15caa-bc11-4a6a-8998-ea0c69df8b9d/TSTFI0211073.bam'))
+])
+def test_extract_bucket_and_key(url, expected):
+    """ Tests the helper function for pulling bucket/key information from a URL """
+    bucket, key = extract_bucket_and_key(url)
+    assert bucket == expected[0]
+    assert key == expected[1]
+
+
+def _test_uri_get(es_testapp, uri):
+    """ Helper functions that tests that we can get back download creds via GET """
+    res = es_testapp.get(f'/download_cli/?item={uri}').json['upload_credentials']
+    assert 'AccessKeyId' in res
+    assert 'ASIA' in res['AccessKeyId']  # only real check we can do that this key is real
+    assert 'SecretAccessKey' in res
+    assert 'SessionToken' in res
+
+
+def _test_uri_post(es_testapp, uri):
+    """ Helper functions that tests that we can get back download creds via POST """
+    res = es_testapp.post_json('/download_cli/', {
+        'item': f'{uri}'
+    }).json['upload_credentials']
+    assert 'AccessKeyId' in res
+    assert 'ASIA' in res['AccessKeyId']  # only real check we can do that this key is real
+    assert 'SecretAccessKey' in res
+    assert 'SessionToken' in res
+
+
+@pytest.mark.workbook
+def test_download_cli_workbook_post(workbook, es_testapp):
```

Comment:
Consider adding a test case for behavior when provided an item that lacks an `@@download` endpoint.